src/oemof/solph/components/_source.py

```diff
@@ -19,10 +19,42 @@
 
 
 class Source(on.Source):
-    """An object with one output flow."""
+    """A component which is designed for one output flow.
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    Parameters
+    ----------
+    label : str
+        String holding the label of the Source object.
+        The label of each object must be unique.
+
+    Examples
+    --------
+    Defining a Source:
+
+    >>> from oemof import solph
+    >>> bel = solph.buses.Bus(label='electricity')
+
+    >>> pv_plant = solph.components.Source(
+    ...    label='pp_pv',
+    ...    outputs={bel: solph.flows.Flow()})
+
+    >>> type(pv_plant)
+    <class 'oemof.solph.components._source.Source'>
+
+    >>> str(pv_plant.outputs())
+    ['electricity']
+
+    Notes
+    -----
+    It is theoretically possible to use the Source object with multiple outputs.
+    However, we strongly recommend using multiple Source objects instead.
+    """
+
+    def __init__(self, label=None, outputs=None):
+        if outputs is None:
+            outputs = {}
+
+        super().__init__(label=label, outputs=outputs)
         check_node_object_for_missing_attribute(self, "outputs")
```

Comment:
Function needs to be replaced by refactored function that raises error message and should be triggered in if-loop (line 55)